datafusion/core/src/dataframe.rs

```diff
@@ -930,10 +932,11 @@ impl DataFrame {
         self,
         path: &str,
         writer_properties: Option<WriterProperties>,
+        save_mode: FileWriterSaveMode,
```

Comment:
Spark made the save mode optional, we can also default to `append` mode. Also, we can merge the `WriterProperties` and `FileWriterSaveMode` structs like in a `ParquetWriterOptions` struct since they are both configurations, this would abstract the save mode from immediate usage and contribute the simplicity.

Reply:
That is good point, alternatively for parquet we can reuse `WriterProperties::key_value_metadata` to specify optional saveMode and default to some value in case of None

Reply:
I saw that option, however, it would better to be have a general `***WriterOptions`, like we have `ParquetReadOptions`, `CsvReadOptions`, `AvroReadOptions` etc.

Reply:
> I saw that option, however, it would better to be have a general ***WriterOptions, like we have ParquetReadOptions, CsvReadOptions, AvroReadOptions etc.

I agree that making a more future proof API here would be very helpful  -- for example, I can imagine adding a other (DataFusion) specific properties for writing parquet. 

For example, I would love to have a place to put "sort expressions" to sort the output parquet file by some set of expressions. 

I love @metesynnada 's suggestion for `ParquetWriterOptions`

Reply:
I agree with @alamb , a wrapper for options might be more future proof.